modules/sdk-coin-sol/src/lib/stakingRawMsgAuthorizeBuilder.ts

```
@@ -0,0 +1,129 @@
+   * @returns {StakeBuilder} This staking builder.
+   *
+   */
+  transactionMessage(msg: string): this {
+    this._transactionMessage = msg;
+    return this;
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    assert(this._transactionMessage, 'missing transaction message');
+
+    this.transaction.solTransaction = SOLTransaction.populate(
+      SOLMessage.from(Buffer.from(this._transactionMessage, 'base64')),
+      []
+    );
+    this.transaction.setTransactionType(this.transactionType);
+    return this.transaction;
+  }
+
+  validateTransaction(tx: Transaction): boolean {
+    return this.validateMessage(tx.solTransaction.serializeMessage().toString('base64'));
+  }
+
+  async build(): Promise<BaseTransaction> {
+    this.validateMessage(this._transactionMessage);
+    return this.buildImplementation();
+  }
+
+  protected validateMessage(msg: string): boolean {
+    const tx = SOLTransaction.populate(SOLMessage.from(Buffer.from(msg, 'base64')), []);
+    const instructions = tx.instructions;
+    assert(instructions.length === 2, 'Invalid transaction, expected 2 instruction, got ' + instructions.length);
+    for (const instruction of instructions) {
+      switch (instruction.programId.toString()) {
+        case SystemProgram.programId.toString():
+          const instructionName = SystemInstruction.decodeInstructionType(instruction);
+          assert(instructionName === 'AdvanceNonceAccount', 'Invalid transaction:' + instructionName);
+          break;
+        case StakeProgram.programId.toString():
+          const data = instruction.data.toString('hex');
+          assert(data === '0a00000001000000', 'Invalid staking instruction data:' + data);
+          break;
+        default:
+          assert(
+            false,
+            'Invalid transaction, instruction program id not supported: ' + instruction.programId.toString()
+          );
```

You can just throw an error, asserting a constant false value is not very intuitive. 

```suggestion
	throw new Error(`Invalid transaction, instruction program id not supported: ${instruction.programId.toString()}`);
```